File: ice0208/prefix-sum/1732.py
class Solution:
    def largestAltitude(self, gain: list[int]) -> int:
        prefix_sum = 0
        max_sum = 0

        for g in gain:
            prefix_sum += g
            max_sum = max(max_sum, prefix_sum)

        return max_sum


if __name__ == "__main__":
    print(Solution().largestAltitude(gain=[-5, 1, 5, 0, -7]))
    print(Solution().largestAltitude(gain=[-4, -3, -2, -1, 4, 3, 2]))

File: moonpoison/prefix_sum/1732.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int largestAltitude(vector<int>& gain) 
    {
        vector<int> v;
        v.push_back(0);
        int ans=0;
        for(int i=0;i<gain.size();++i)
        {
            v.push_back(v[i]+gain[i]);
            if(ans<v[i+1]) ans=v[i+1];
        }
        return ans;
    }
};